core/src/main/java/com/orientechnologies/orient/core/index/OIndexManagerShared.java

```diff
@@ -556,44 +545,62 @@ private void recreateIndexes(OStorage storage, Collection<ODocument> idxs, OData
       OLogManager.instance().info(this, "%d indexes were restored successfully, %d errors", ok, errors);
     }
 
-    private void recreateIndex(OStorage storage, ODocument idx) {
-      final OIndexInternal<?> index = createIndex(idx);
-      final OIndexMetadata indexMetadata = index.loadMetadata(idx);
+    private void recreateIndex(ODocument indexDocument) {
+      final OIndexInternal<?> index = createIndex(indexDocument);
+      final OIndexMetadata indexMetadata = index.loadMetadata(indexDocument);
       final OIndexDefinition indexDefinition = indexMetadata.getIndexDefinition();
 
-      if (indexDefinition != null && indexDefinition.isAutomatic()) {
-        // XXX: At this moment Lucene-based indexes are not durable, so we still need to rebuild them.
-        if ("LUCENE".equalsIgnoreCase(indexMetadata.getAlgorithm())) {
-          try {
-            index.loadFromConfiguration(idx);
-            index.delete();
-          } catch (Exception e) {
-            OLogManager.instance()
-                .error(this, "Error on removing index '%s' on rebuilding. Trying removing index files (Cause: %s)", index.getName(),
-                    e);
-
-            // TRY DELETING ALL THE FILES RELATIVE TO THE INDEX
-            for (Iterator<OIndexFactory> it = OIndexes.getAllFactories(); it.hasNext(); ) {
-              try {
-                final OIndexFactory indexFactory = it.next();
-                final OIndexEngine engine = indexFactory.createIndexEngine(null, index.getName(), false, storage, 0, null);
-
-                engine.deleteWithoutLoad(index.getName());
-              } catch (Exception e2) {
-              }
-            }
-          }
+      final boolean automatic = indexDefinition != null && indexDefinition.isAutomatic();
+      // XXX: At this moment Lucene-based indexes are not durable, so we still need to rebuild them.
+      final boolean durable = !"LUCENE".equalsIgnoreCase(indexMetadata.getAlgorithm());
 
-          createNonDurableAutomaticIndex(index, indexMetadata, indexDefinition);
-        } else
-          addDurableAutomaticIndex(idx, index, indexMetadata);
+      if (automatic) {
+        if (durable) {
+          OLogManager.instance().info(this, "Index '%s' is a durable automatic index and will be added as is without rebuilding",
+              indexMetadata.getName());
+          addIndexAsIs(indexDocument, index);
+        } else {
+          OLogManager.instance()
+              .info(this, "Index '%s' is a non-durable automatic index and must be rebuilt", indexMetadata.getName());
+          rebuildNonDurableAutomaticIndex(indexDocument, index, indexMetadata, indexDefinition);
+        }
       } else {
-        addNonAutomaticIndex(idx, index, indexMetadata);
+        if (durable) {

```

Comment:
Why we re-add already existing indexes if they already in index manager, please either provide clear comment in code or delete this code